File: Java/518. Coin Change II.java
class Solution {
    public int change(int amount, int[] coins) {
        int n = coins.length;
        int[] dp = new int[amount+1];
        dp[0]=1;
        for(int i = 0 ; i<n;i++){
            for(int j=1;j<=amount;j++){
                if(j-coins[i]>=0){
                    dp[j]+=dp[j-coins[i]];
                }
            }
        }
        return dp[amount];   
    }
}

File: JavaScript/518. Coin Change II.js
/**
 * @param {number} amount
 * @param {number[]} coins
 * @return {number}
 */
var change = function(amount, coins) {
    var n = coins.length;
    var dp = new Array(amount+1).fill(0);
    dp[0]=1;
    for(var i = 0 ; i<n;i++){
        for(var j=1;j<=amount;j++){
            if(j-coins[i]>=0){
                dp[j]+=dp[j-coins[i]];
            }
        }
    }
    return dp[amount];   
};
